### PR TITLE
Great homework

### PR DESCRIPTION
**Pull request by iqusaii**

### CODE REVIEW

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml [hunk 1]
```diff
@@ -1,9 +1,68 @@
...
+    r1.computeArea() //call the fun from the sub class
+    t1.computeArea() //call the fun from the sub class
+
+
+}
+open class Shape { //create class
+    val shapeName =" " //declaration of variable
+    val shapeColor = " " //declaration of variable
+    open fun info(color:String,shape:String): String { //create fun
+        return color+shape // return value
```
Comment (shuruqt1000) [inline .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:16]: why use open for function and class ? what dose it mean ?

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+    }
+}
+
+//file2
+class Circle:Shape() { //inherit the super class
+    val pi = 3.14
+    val radius = 0
+
+     override fun computeArea() { //override the fun
+    println("Enter radius")
```
Comment (shuruqt1000) [inline .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:30]: why use override ? what dose it change exactly ?what effect dose it have on the rest of the code ?

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml [hunk 3]
```diff
@@ -1,9 +1,68 @@
...
+}
+//file3
+open class Rectangular:Shape() { //inherit the super class
+    val width =0
+    val height = 0
+    override fun computeArea() { //override the fun
+        println("Enter th height")
+        var h = readLine()?.toDouble()//take the height from user
+        println("enter the width")
+        var w = readLine()?.toDouble()//take the width from user
```
Comment (shuruqt1000) [inline .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:45]: why open ? why use override ? what dose it change exactly ?what effect dose it have on the rest of the code

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml [hunk 4]
```diff
@@ -1,9 +1,68 @@
...
+
+    }
+}
+//file4
+open class Triangle:Shape() { //inherit the super class
+    val base =0
+    val height = 0
+    override fun computeArea() { //override the fun
+        println("Enter th height")
+        var h = readLine()?.toDouble()//take the height from user
```
Comment (shuruqt1000) [inline .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:56]: why use override ? what dose it change exactly ?what effect dose it have on the rest of the code

**Review by shuruqt1000 — CHANGES_REQUESTED** [review]
if there is any values passed from where and why please explain your code (not restate what it says)

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml (hunk 3 above)
Comment (iqusaii) [reply .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:45]: to change the original fun

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml (hunk 4 above)
Comment (iqusaii) [reply .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:56]: to change the original fun

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml (hunk 2 above)
Comment (iqusaii) [reply .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:30]: to change the original fun

File: .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml (hunk 1 above)
Comment (iqusaii) [reply .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:16]: to allow us to change it when we inherit it

Comment (iqusaii) [reply .idea/HW-Week2_Day03_Inheritance_and_ClassTypes.iml:16]: to allow us to change it when we inherit it

**Comment by shuruqt1000** [comment]
good comments 
completed

**Pull request closed without merging**